prefect_aws/s3.py

```diff
@@ -9,13 +9,15 @@
 from prefect import get_run_logger, task
 
 from prefect_aws import AwsCredentials
+from prefect_aws.client_parameters import AwsClientParameters
 
 
 @task
 async def s3_download(
     bucket: str,
     key: str,
     aws_credentials: AwsCredentials,
+    aws_client_parameters: AwsClientParameters = AwsClientParameters(),
```

Comment:
I'm wondering if it's better to set this to None; then inside the code block check if None.

Reply:
Also do we need a separate dataclass for this or simply **aws_client_kwargs? The benefit is docstrings within a dataclass

Reply:
> I'm wondering if it's better to set this to None; then inside the code block check if None.

What is the advantage of passing None as default vs the default class ?

Reply:
> Also do we need a separate dataclass for this or simply **aws_client_kwargs? The benefit is docstrings within a dataclass

We talked about that with @desertaxle in issue #25. We thought that using a strong dataclass would be better.

Reply:
Aha I wasn't following that; in that case yeah, go with a dataclass!

In regards to an instantiated class, from https://medium.com/python-features/how-to-avoid-classic-pitfall-while-passing-default-values-in-python-7002c0dc4c7c#:~:
```
Do not use mutable default arguments in Python. “mutable” refers to anything such as a list, dictionary or even a class instance.
Use None as the default value for keyword arguments and assign the mutable value inside the function.
```

Reply:
Yes, that makes sense, however here I want to set up the value with this default empty ```AwsClientParameters```. is not something that will change as default. ```AwsClientParameters``` is flagged as frozen so we won't have any risk of being mutated.
Happy to change if you want. however I don't think this is required here